app/websocket_app.rb

```diff
@@ -49,6 +49,16 @@ def call(message)
       job_id = message[:job_id]
       FlightScheduler.app.event_processor.node_failed_job(@node_name, job_id)
 
+    when 'NODE_COMPLETED_ARRAY_TASK'
+      task_id = message[:array_task_id]
+      job_id = message[:array_job_id]
```

Comment:
In general we should be able to infer the `job_id` from the `task_id` as they exist in a one-many relationship. Looks like we need both due to the data model.

This is fine for now as we can deprecate the `job_id` without introducing a breaking change.

Reply:
At the moment, we can't sensibly retrieve an array task from its id.  Once we have job/task persistence in place we'll be able to do so.  At that point, sending only the `task_id` might be the better option.

Regarding, deprecation and breaking changes, we don't need to concern ourselves with that whilst the project is still in alpha.  We just need to make sure that our three `master` branches work together correctly.

Reply:
:+1: